### PR TITLE
[PATCH v4] linux-gen: ml: a few small fixes and cleanups

### PR DESCRIPTION
**Pull request by JereLeppanen**

### CODE REVIEW

**Comment by JereLeppanen** [comment]
v2:
- Rebase.
- Check for null pointers in `linux-gen: ml: slightly simplify clean up in error cases in odp_ml_run()`
- Add commit `linux-gen: ml: clean up error handling in verify_tensor()`

**Review by MatiasElo — APPROVED** [review]
Copyright can be updated.

**Comment by JereLeppanen** [comment]
v3:
- Update copyright in first commit.
- Add review tags.

File: platform/linux-generic/odp_ml.c [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /* SPDX-License-Identifier: BSD-3-Clause
- * Copyright (c) 2023 Nokia
+ * Copyright (c) 2024 Nokia
```
Comment (MatiasElo) [inline platform/linux-generic/odp_ml.c:2]: The style has been:`2023-2024`

Comment (JereLeppanen) [reply platform/linux-generic/odp_ml.c]: Fixed.

**Pull request merged**